features/git-rename-branch/non_feature_branch/with_remote_origin/with_tracking_branch/with_unpushed_changes.feature

```diff
@@ -0,0 +1,34 @@
+Feature: git rename-branch: errors if renaming a non-feature branch that has unpushed changes
+
+  As a developer renaming a non-feature branch that has unpushed changes
+  I should get an error that the given branch is not in sync with its tracking branch
+  So that I don't lose work by accidentally overwriting existing branches.
```

Comment:
Yeah, I suppose it would just sync the branch. We could implement some detection for only unpushed changes, but I think it would only be needed until syncing is added, so I don't know if it's worth it. What do you think of a correction to `So that I know the branch must be in sync` or similar?